drivers/stmpe811/stmpe811_touch_dev.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2020 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_stmpe811
+ * @{
+ *
+ * @file
+ * @brief       Driver adaption to touch_dev generic interface
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ * @}
+ */
+
+#include <stddef.h>
+#include <stdint.h>
+#include <errno.h>
+#include <assert.h>
+
+#include "kernel_defines.h"
+
+#include "stmpe811.h"
+#include "stmpe811_touch_dev.h"
+
+uint16_t _stmpe811_height(const touch_dev_t *touch_dev)
+{
+    const stmpe811_t *dev = (stmpe811_t *)touch_dev;
+    assert(dev);
+
+    return dev->params.ymax;
+}
+
+uint16_t _stmpe811_width(const touch_dev_t *touch_dev)
+{
+    const stmpe811_t *dev = (stmpe811_t *)touch_dev;
+    assert(dev);
+
+    return dev->params.xmax;
+}
+
+uint8_t _stmpe811_touches(const touch_dev_t *touch_dev, touch_t *touches, size_t len)
+{
+    stmpe811_t *dev = (stmpe811_t *)touch_dev;
+    assert(dev);
+
+    /* stmpe811 is only single touch */
+    assert(len == 1);
```

Comment:
Is this check really needed? If an application uses this interface to have device-agnostic access to the touch sensor it might pass always a pre-allocated array.

Reply:
Ok, then removed !